src/engine/ExtendedArrayMesh.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Godot;
+
+/// <summary>
+///   Wrapper of an <see cref="ArrayMesh"/> with additional functionality
+/// </summary>
+public class ExtendedArrayMesh
```

Comment:
On reviewing how this is used, this seems like a totally unnecessary thing. The unwrap method just adjusts the internal ArrayMesh, so a freestanding C++ method exposed through the interop way should be able to perform the exact same operation without needing to allocate an extra object instance.

Reply:
Would `mesh.NativeInstance` be the right way to get an ArrayMesh's pointer in C#?

Reply:
I'm not sure such a property exists, but internally godot calls `GodotObject.GetPtr(mesh)` to get the native instance pointer when it is calling the C# bound methods. We *probably* need to use a bit of reflection to get into the godot internals to be able to grab the pointer.

Reply:
This field does exist, and ultimately it seems to do the same as `GetPtr()`:
`public IntPtr NativeInstance => NativePtr;`

However, the C++ code crashes when called with such a pointer. From using some debug returns, I can conclude that it crashes when a mesh's function is called. 
So, it seems like there's some kind of issue with the pointer passed.

Reply:
Could it be that the pointers passed to the C# side are adjusted somehow? That's about the only thing I can think off. It should be possible to use a C++ debugger to add a breakpoint in the extension and then inspect the passed in pointer as if it was a mesh to see if it is correct or not. If the pointer is not good then I see just three possibilities: the pointer access in C# side is wrong, the Godot engine does some pointer offsetting that are passed to/from C# so we need to read the godot source code to find out how to undo that ourselves, or I think you can script gdb for example to just try incrementing and decrementing the pointer to see if any small-ish offset results in valid data appearing.